frontend/src/metabase/visualizations/echarts/pie/tooltip.tsx

```diff
@@ -33,16 +33,25 @@ export const getTooltipOption = (
     ...getTooltipBaseOption(containerRef),
     trigger: "item",
     formatter: params => {
-      if (Array.isArray(params) || typeof params.dataIndex !== "number") {
+      if (Array.isArray(params) || !params.name) {
+        return "";
+      }
+
+      const slice =
```

Comment:
It is a corner case, but would be nice to fix by switching from using slice names as identifiers:
https://github.com/user-attachments/assets/de74a074-c6ec-4909-89fd-19a1c01679d2

Reply:
Good catch, thanks! [`56036a8`](https://github.com/metabase/metabase/pull/47581/commits/56036a8b398edb5d69c69ec8f1d5b3536f30e29a) should fix it, I found a way to make it work using the `dataIndex`

Also, I noticed that if there's a real pie slice called "Other" + the grouped "Other" slice, the toggle behavior wasn't working as expected. The problem was that we use the `slice.data.key` property to keep the list of hidden slices, which happened to be the same in these two scenarios. In [`17a4c1f`](https://github.com/metabase/metabase/pull/47581/commits/17a4c1fdc2b481d807f025b25a980a8193ea6e98) I've reworked it so the grouped "Other" slice's key is `___OTHER___` (cc @EmmadUsmani)